core/slice.go

```diff
@@ -287,6 +296,73 @@ func (sl *Slice) updateCacheAndRelay(pendingHeader types.PendingHeader, location
 	}
 }
 
+// CollectEtxsForManifest will gather the full list of ETXs that are referencable through a given manifest
+func (sl *Slice) CollectEtxsForManifest(manifest types.BlockManifest) (types.Transactions, error) {
+	etxs := types.Transactions{}
+	for _, hash := range manifest {
+		res, ok := sl.pendingEtxs.Get(hash)
+		if !ok || res == nil {
+			return nil, fmt.Errorf("unable to find pending etxs for hash in manifest", "hash: ", hash)
+		}
+		pendingEtxs := res.([]types.Transactions)
+		etxs = append(etxs, pendingEtxs[common.PRIME_CTX]...)
+		etxs = append(etxs, pendingEtxs[common.REGION_CTX]...)
+		etxs = append(etxs, pendingEtxs[common.ZONE_CTX]...)
+	}
+	return etxs, nil
+}
+
+// CollectNewlyConfirmedEtxs collects all newly confirmed ETXs since the last coincident with the given location
+func (sl *Slice) CollectNewlyConfirmedEtxs(block *types.Block, location common.Location) (types.Transactions, error) {
+	nodeCtx := common.NodeLocation.Context()
+	// Collect all ETXs referenced through the manifest. These are now spendable
+	// and should be made available both locally and to our subordinate.
+	referencableEtxs, err := sl.CollectEtxsForManifest(block.SubManifest())
+	if err != nil {
+		return nil, err
+	}
+	referencableEtxs = append(referencableEtxs, block.ExtTransactions()...) // Include ETXs emitted in this block
+
+	// Filter for ETXs destined to this slice
+	newInboundEtxs := referencableEtxs.FilterToSlice(location, nodeCtx)
+
+	// Filter this list to exclude any ETX for which we are not the crossing
+	// context node. Such ETXs cannot be used by our subordinate for one of the
+	// following reasons:
+	// * if we are prime, but common dom was a region node, than the given ETX has
```

Comment:
I am confused with the use of common dom here. Since you use common dom for prime.

Reply:
This is one of those things that probably needs a diagram to explain effectively. I'll attempt to explain in text (and maybe update the comment?), but tomorrow we can draw it out IRL and I can attempt to translate those into diagrams for the spec doc.

Here I say "common dom" of two chains to describe the highest context chain which exists in each chain's slice. e.g:
* zone-0-0 & zone-0-1 would share region-0 as their highest context common dom chain
* zone-0-0 & zone-1-0 would share Prime as their highest context common dom

This is effectively the "crossing context" for a cross-chain transaction. The ETX must become referencable in the "common dom"s context, before it can become referencable in the destination chain. After an ETX becomes referencable in the common dom's context, then those ETXs will become minable at the next coincident block with the destination.

Reply:
lmk how I can update this commend and/or the cross-chain spec to clarify this, because this is the crux of the entire mechanism.

Reply:
This makes sense. I think just a definition of `common dom` is good enough. you can understand everything else.